go/vt/vttablet/tabletserver/tx_engine_test.go

```diff
@@ -625,43 +625,43 @@ func TestCheckReceivedError(t *testing.T) {
 	}{{
 		receivedErr: vterrors.New(vtrpcpb.Code_DEADLINE_EXCEEDED, "deadline exceeded"),
 		retryable:   true,
-		expQuery:    `update _vt.redo_state set state = 1, message = 'deadline exceeded' where dtid = 'aa'`,
+		expQuery:    `update _vt.redo_state set state = 1, message = 'deadline exceeded' where dtid = _binary'aa'`,
 	}, {
 		receivedErr: vterrors.New(vtrpcpb.Code_INVALID_ARGUMENT, "invalid argument"),
 		retryable:   false,
-		expQuery:    `update _vt.redo_state set state = 0, message = 'invalid argument' where dtid = 'aa'`,
+		expQuery:    `update _vt.redo_state set state = 0, message = 'invalid argument' where dtid = _binary'aa'`,
 	}, {
 		receivedErr: sqlerror.NewSQLError(sqlerror.ERLockDeadlock, sqlerror.SSLockDeadlock, "Deadlock found when trying to get lock; try restarting transaction"),
 		retryable:   false,
-		expQuery:    `update _vt.redo_state set state = 0, message = 'Deadlock found when trying to get lock; try restarting transaction (errno 1213) (sqlstate 40001)' where dtid = 'aa'`,
+		expQuery:    `update _vt.redo_state set state = 0, message = 'Deadlock found when trying to get lock; try restarting transaction (errno 1213) (sqlstate 40001)' where dtid = _binary'aa'`,
```

Comment:
Is this the side effect or it is necessary to have `_binary` for all varbinary column types?

Reply:
This query is generated with a bind var which now always uses _binary for a binary value. 

Doing this is required to fix the original bug where variables weren't correctly typed and resulted in broken behavior.